frontend/src/components/Buttons/FilterButton.vue

```diff
@@ -292,12 +307,7 @@ function emitFilterChange(): void {
     years: selectedYearFilters.value
   });
 }
-watch(
-  () => props.itemsType,
-  () => {
-    refreshItems();
-  }
-);
+watch(() => props.item, refreshItems);
```

Comment:
The logic behind the library/filter/sort buttons have always seemed so tangled to me that I never tried to understood it properly (not your fault ofc, in fact you're brave by trying to fix past messes!)
However, better be safe than worry: are you sure we don't need to call ``refreshItems`` at mount? It's truly only needed when filters are applied?

In case you discover we need it at mount, make this watcher immediate please.